src/api/spec/models/watched_item_spec.rb

```diff
@@ -0,0 +1,185 @@
+
+    context 'when the user tries to add the same project twice to the watchlist' do
+      let(:project) { create(:project) }
+      let(:user) { create(:user) }
+      let(:watched_item) { build(:watched_item, :for_projects, watchable: project, user: user) }
+
+      before { create(:watched_item, :for_projects, watchable: project, user: user) }
+
+      it 'is not valid' do
+        expect(watched_item).not_to be_valid
+        expect(watched_item.errors.full_messages).to eql(["Watchable has already been taken"])
+      end
+    end
+
+    context 'when the user tries to add different projects to the watchlist' do
+      let(:watched_item) { build(:watched_item, :for_projects) }
+
+      before { create(:watched_item, :for_projects) }
+
+      it 'is valid' do
+        expect(watched_item).to be_valid
+      end
+    end
+    
+    context 'when two different users try to watch the same project' do
+      let(:watched_item) { create(:watched_item, :for_projects) }
+
+      before { create(:watched_item, :for_projects) }
+
+      it 'is valid' do
+        expect(watched_item).to be_valid
+      end
+    end
+  end
+
+  context 'when watching a package' do
+    let(:package) { create(:package) }
+    let(:watched_item) { build(:watched_item, :for_packages) }
+    
+    it 'is valid with valid attributes' do
+      expect(watched_item).to be_valid
+    end
+    
+    context 'when missing a package' do
+      it 'is not valid' do
+        watched_item = build(:watched_item, :for_packages, watchable: nil)
+        expect(watched_item).not_to be_valid
+        expect(watched_item.errors.full_messages).to include("Watchable must exist",
+                                                            "Watchable type can't be blank",
+                                                            "Watchable can't be blank")
+      end
+    end
+    
+    context 'when missing a user' do
+      let(:watched_item) { build(:watched_item, :for_packages, user: nil) }
+
+      it 'is not valid' do
+        expect(watched_item).not_to be_valid
+        expect(watched_item.errors.full_messages).to eql(["User must exist"])
+      end
+    end
+
+    context 'when the user tries to add the same package twice to the watchlist' do
+      let(:package) { create(:package) }
+      let(:user) { create(:user) }
+      let(:watched_item) { build(:watched_item, :for_packages, watchable: package, user: user) }
+
+      before { create(:watched_item, :for_packages, watchable: package, user: user) }
+
+      it 'is not valid' do
+        expect(watched_item).not_to be_valid
+        expect(watched_item.errors.full_messages).to eql(["Watchable has already been taken"])
+      end
+    end
+
+    context 'when the user tries to add different packages to the watchlist' do
+      let(:watched_item) { build(:watched_item, :for_packages) }
+
+      before { create(:watched_item, :for_packages) }
+
+      it 'is valid' do
+        expect(watched_item).to be_valid
+      end
+    end
```

Comment:
Same. I think specifying the same **user** in both factory creations is missing here :thinking: